code/bignum/Hacl.Impl.Exponentiation.fsti

```diff
@@ -0,0 +1,209 @@
+module Hacl.Impl.Exponentiation
+
+open FStar.HyperStack
+open FStar.HyperStack.ST
+open FStar.Mul
+
+open Lib.IntTypes
+open Lib.Buffer
+
+module ST = FStar.HyperStack.ST
+module B = LowStar.Buffer
+
+module LSeq = Lib.Sequence
+module BSeq = Lib.ByteSequence
+module Loops = Lib.LoopCombinators
+
+module S = Lib.Exponentiation
+module BD = Hacl.Bignum.Definitions
+
+#reset-options "--z3rlimit 50 --fuel 0 --ifuel 0"
+
+inline_for_extraction noextract
+let inttype_a = t:inttype{t = U32 \/ t = U64}
+
+inline_for_extraction noextract
+class lexp_to_exp (a_t:inttype_a) (len:size_t{v len > 0}) (ctx_len:size_t) = {
+  a_spec: Type0;
+  exp: S.exp a_spec;
+  linv_ctx: x:LSeq.lseq (uint_t a_t SEC) (v ctx_len) -> Type0;
+  linv: x:LSeq.lseq (uint_t a_t SEC) (v len) -> Type0;
+  refl: x:LSeq.lseq (uint_t a_t SEC) (v len){linv x} -> GTot a_spec;
+}
+
+
+// inline_for_extraction noextract
```

Comment:
That would be useful no?

Reply:
not really, it is easier to initialize the resulting buffer with `one` before calling the exponentiation function.
for example, we can convert an integer to its montgomery representation using the precomputed constant `r2`. 
this means that we either need to store `r2` as a part of the `ctx` buffer or compute it internally (which is quite expensive).